nestkernel/conn_builder.cpp

```diff
@@ -74,75 +71,145 @@ nest::ConnBuilder::ConnBuilder( NodeCollectionPTR sources,
   updateValue< bool >( conn_spec, names::make_symmetric, make_symmetric_ );
 
   // read out synapse-related parameters ----------------------
-  if ( not syn_spec->known( names::synapse_model ) )
-  {
-    throw BadProperty( "Synapse spec must contain synapse model." );
-  }
-  const std::string syn_name = ( *syn_spec )[ names::synapse_model ];
-  if ( not kernel().model_manager.get_synapsedict()->known( syn_name ) )
+
+  // synapse-specific parameters
+  std::set< Name > skip_set = {
+    names::weight, names::delay, names::min_delay, names::max_delay, names::num_connections, names::synapse_model
+  };
+
+  default_weight_.resize( syn_spec.size() );
+  default_delay_.resize( syn_spec.size() );
+  default_weight_and_delay_.resize( syn_spec.size() );
+  weight_.resize( syn_spec.size() );
+  delay_.resize( syn_spec.size() );
+  synapse_params_.resize( syn_spec.size() );
+  synapse_model_id_.resize( syn_spec.size() );
+  synapse_model_id_[ 0 ] = kernel().model_manager.get_synapsedict()->lookup( "static_synapse" );
+  param_dicts_.resize( syn_spec.size() );
+
+  int indx = 0;
```

Comment:
Is `indx` the best name? If I see it right, then `indx` moves in lock-step with `syn_params`? So in Python we could just write `for indx, syn_p in enumerate(syn_spec)`. I fear C++ doesn't quite give us that, but I think we should make the coupling more explicit, e.g.
```c++
for ( size_t indx = 0 ; indx < syn_spec.size() ; ++indx )
{
  auto syn_params = syn_spec[indx];
  ...
```